lib/huginn_agent.rb

```diff
@@ -1,17 +1,45 @@
 require 'huginn_agent/version'
 
 class HuginnAgent
-  def self.register(*paths)
-    @paths ||= []
-    paths.each do |path|
-      @paths << path
+  class << self
+    attr_accessor :branch, :remote
+
+    def load_tasks(options = {})
+      @branch = options[:branch] || 'master'
+      @remote = options[:remote] || 'https://github.com/cantino/huginn.git'
+      Rake.add_rakelib File.join(File.expand_path('../', __FILE__), 'tasks')
+    end
+
+    def load(*paths)
+      paths.each do |path|
+        load_paths << path
+      end
+    end
+
+    def register(*paths)
```

Comment:
Rather than providing paths to look for agents, could we structure this such that Agents "register" themselves as they're loaded? Something like:
```ruby
class MyAgent < HuginnAgent
  add_self_to_registry
  ...
end

class HuginnAgent
  cattr_accessor :agent_types
  agent_types = []
  def self.add_self_to_registry
    agent_types |= [self]
  end
  ...
end
```

Or, per cantino/huginn#1526 just call `HuginnAgent.descendants`